locations/jclouds/src/test/java/org/apache/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizerTest.java

```diff
@@ -147,6 +149,59 @@ public void testAddPermissionsToNode() {
     }
 
     @Test
+    public void testRemovePermissionsFromNode() {
+        IpPermission ssh = newPermission(22);
+        IpPermission jmx = newPermission(31001);
+        String nodeId = "node";
+        SecurityGroup sharedGroup = newGroup(customizer.getNameForSharedSecurityGroup());
+        SecurityGroup group = newGroup("id");
+        when(securityApi.listSecurityGroupsForNode(nodeId)).thenReturn(ImmutableSet.of(sharedGroup, group));
+        when(computeService.getContext().unwrap().getId()).thenReturn("aws-ec2");
+
+        customizer.addPermissionsToLocation(ImmutableList.of(ssh, jmx), nodeId, computeService);
+        customizer.removePermissionsFromLocation(ImmutableList.of(jmx), nodeId, computeService);
+
+        verify(securityApi, never()).removeIpPermission(ssh, group);
+        verify(securityApi, times(1)).removeIpPermission(jmx, group);
+    }
+
+    @Test
+    public void testRemoveMultiplePermissionsFromNode() {
+        IpPermission ssh = newPermission(22);
+        IpPermission jmx = newPermission(31001);
+        String nodeId = "node";
+        SecurityGroup sharedGroup = newGroup(customizer.getNameForSharedSecurityGroup());
+        SecurityGroup group = newGroup("id");
+        when(securityApi.listSecurityGroupsForNode(nodeId)).thenReturn(ImmutableSet.of(sharedGroup, group));
+        when(computeService.getContext().unwrap().getId()).thenReturn("aws-ec2");
+
+        customizer.addPermissionsToLocation(ImmutableList.of(ssh, jmx), nodeId, computeService);
+        customizer.removePermissionsFromLocation(ImmutableList.of(ssh, jmx), nodeId, computeService);
+
+        verify(securityApi, times(1)).removeIpPermission(ssh, group);
+        verify(securityApi, times(1)).removeIpPermission(jmx, group);
+    }
+
+
+    @Test
+    public void testAddPermissionWhenNoExtension() {
+        IpPermission ssh = newPermission(22);
+        IpPermission jmx = newPermission(31001);
+        String nodeId = "node";
+
+        when(securityApi.listSecurityGroupsForNode(nodeId)).thenReturn(Collections.<SecurityGroup>emptySet());
+
+        RuntimeException exception = null;
+        try {
+            customizer.addPermissionsToLocation(ImmutableList.of(ssh, jmx), nodeId, computeService);
```

Comment:
The approach we generally take is to `fail("should have thrown some exception")` after the exceptional statement and then to assert on the exception in the catch block. You can also use `Exceptions.getFirstInteresting(e)` and the recent additions of `shouldHaveFailedPreviously()` and `expectedFailure()` to the `Asserts` class.